internal/resource/config/serversettings/server_settings_resource.go

```diff
@@ -981,21 +982,17 @@ func readServerSettingsResponse(ctx context.Context, r *client.ServerSettings, s
 	//////////////////////////////////////////////
 	// CAPTCHA SETTINGS
 	//////////////////////////////////////////////
-	var getCaptchaSettingsAttrValue = func() map[string]attr.Value {
-		if internaltypes.ObjContainsNoEmptyVals(plan.CaptchaSettings) {
-			return map[string]attr.Value{
-				"site_key":   types.StringPointerValue(r.CaptchaSettings.SiteKey),
-				"secret_key": types.StringValue(plan.CaptchaSettings.Attributes()["secret_key"].(types.String).ValueString()),
-			}
-		} else {
-			return map[string]attr.Value{
-				"site_key":   types.StringPointerValue(&emptyString),
-				"secret_key": types.StringValue(emptyString),
-			}
+	if internaltypes.IsDefined(plan.CaptchaSettings) {
+		captchaSettingsAttrValue := map[string]attr.Value{
+			"site_key":   types.StringPointerValue(r.CaptchaSettings.SiteKey),
+			"secret_key": types.StringValue(plan.CaptchaSettings.Attributes()["secret_key"].(types.String).ValueString()),
```

Comment:
What happens if secret_key isn't set in the captcha settings? Looks like it's optional in the schema.